File: src/main/java/org/cubewhy/celestial/game/RemoteAddon.java
/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */

/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */
package org.cubewhy.celestial.game;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

import java.net.URL;

@AllArgsConstructor
@Getter
public class RemoteAddon {
    String name;
    URL downloadURL;
    Category category;

    public enum Category {
        AGENT,
        CN,
        WEAVE;


        /**
         * Parse plugin type from a string
         * */
        @Contract(pure = true)
        public static Category parse(@NotNull String category) {
            return switch (category) {
                case "cn" -> CN;
                case "weave", "Mod" -> WEAVE;
                case "Agent" -> AGENT;
                default -> null;
            };
        }
    }
}

File: src/main/kotlin/org/cubewhy/celestial/game/RemoteAddon.kt
/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */
/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */
package org.cubewhy.celestial.game

import org.jetbrains.annotations.Contract
import java.net.URL

class RemoteAddon(@JvmField var name: String, @JvmField var downloadURL: URL, @JvmField var category: Category) {
    enum class Category {
        AGENT,
        CN,
        WEAVE;


        companion object {
            /**
             * Parse plugin type from a string
             */
            @Contract(pure = true)
            fun parse(category: String): Category? {
                return when (category) {
                    "cn" -> CN
                    "weave", "Mod" -> WEAVE
                    "Agent" -> AGENT
                    else -> null
                }
            }
        }
    }
}
